src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,399 @@
+              ) = InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder).build()
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token)).also(builder)
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token, intent, *intents)).also(builder)
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token, intents)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intent, *intents)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder)
+
+inline fun JDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                     ) =
+    InlineJDABuilder(JDABuilder.create(token, intent, *intents)).also(builder)
+
+inline fun JDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                     ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder)
+
+class InlineJDABuilder(val builder: JDABuilder) {
+    var gatewayEncoding: GatewayEncoding? = null
+        set(value) {
+            if (value != null) {
+                builder.setGatewayEncoding(value)
+            }
+            field = value
+        }
+
+    var rawEventsEnabled: Boolean = false
+        set(value) {
+            builder.setRawEventsEnabled(value)
+            field = value
+        }
+
+    var relativeRateLimit: Boolean = true
+        set(value) {
+            builder.setRelativeRateLimit(value)
+            field = value
+        }
+
+    @Deprecated("We add CacheFlags to the enum over time which will be disabled when using this method. " +
+                        "This introduces breaking changes due to the way the setter works. " +
+                        "enableCache(flags) and disableCache(flags)")
+    var cacheFlags: MutableSet<CacheFlag> = mutableSetOf()
```

Comment:
Why add deprecated things?

Reply:
Originally I was very unsure of how to enable access to the cache flags. I wanted to provide a field for it because that just seemed cleaner to me, but in the end I gave up with that and decided to just have the methods.
I *could* make a custom class that overloads the `plusAssign` and `minusAssign` operators and uses those to add them, though.